src/neuroconv/utils/path_expansion.py

```diff
@@ -0,0 +1,100 @@
+import abc
+import os
+from collections import defaultdict
+from glob import glob
+from typing import Dict, List, Union
+
+from parse import parse
+
+from .types import FilePathType, FolderPathType
+
+
+def _ddict():
+    """
+    Create a defaultdict of defaultdicts
+
+    This allows you to easily nest hierarchical dictionaries. For example, this syntax
+    >>> a = dict(b=dict(c=dict(d=5)))
+
+    becomes
+    >>> a = _ddict()["b"]["c"]["d"] = 5
+
+    It becomes particularly useful when modifying an existing hierarchical dictionary,
+    because the next level is only created if it does not already exist.
+
+    """
+    return defaultdict(_ddict)
```

Comment:
Would you prefer to get #404 in before this so you can use the `DeepDict` class here instead? I'd be in favor of that